fili-core/src/main/java/com/yahoo/bard/webservice/application/ResourceConfig.java

```diff
@@ -39,8 +39,8 @@
      */
     public ResourceConfig() throws ClassNotFoundException, InstantiationException, IllegalAccessException {
         // Build the binder using the configured factory
-        Class<?> binderClass = Class.forName(bindingFactory);
-        BinderFactory binderFactory = (BinderFactory) binderClass.newInstance();
+        Class<? extends BinderFactory> binderClass = Class.forName(bindingFactory).asSubclass(BinderFactory.class);
```

Comment:
This will ensure the class type is valid *before* instantiating then failing to cast.